### DISCUSSION TITLE
issue_id parameter for download_github_comments.rmd

### DISCUSSION THREAD

**Discussion by Ssunoo2** [sailuh/kaiaulu#268] [Q&A] [answered]
When attempting to run this vignette and after making a token, specifying its path, I get this error after line 193:

Error in eval(jsub, SDenv, parent.frame()) : object 'issue_id' not found

I searched the vignette and found that that function was not present, which made it difficult to debug, though I know that issue_id is a parameter inside the specified config file kaiaulu.yml (placeholder is \#[0-9]+). The comment for this parameter says to check the commit message to find out how to make a regex. I was wondering where to find the information required to make a regex for the issue_id parameter.

**Comment by carlosparadis** [comment]
Hi,

For reference, you can left click any line in the repo or multiple holding shift, then a pop-up will show where you can select copy permalink, which will auto-embed and reference the code region in case line 193 changes in the future:

https://github.com/sailuh/kaiaulu/blob/9294a9bb92506b76424b0fcc7c07f6b74d5ce53d/vignettes/download_github_comments.Rmd#L193

Are you referring to this line?

I believe this requires a bit of familiarity with what an issue regex is, which means documentation has to be improved here if you are not familiar with them (commonly used in mining software repositories research). For Kaiaulu, the placeholder should have sufficed, however. 

Line 193 does not seem related to issue_id, so I am a bit confused. Could you double check the line is really that by using the permalink?

**Reply by Ssunoo2** [reply]
Thanks! I did not know I could reference lines with a permalink

https://github.com/sailuh/kaiaulu/blob/9294a9bb92506b76424b0fcc7c07f6b74d5ce53d/vignettes/download_github_comments.Rmd#L193

 replies <- parse_github_replies(save_path) 
Yes, that is the same line. I believe that must be true as I'm unfamiliar with a Regex, so I will try to do some independent research. To confirm, the error I am receiving is

Error in eval(jsub, SDenv, parent.frame()) : object 'issue_id' not found.

This appears in the console as well as in the notebook after the chunk and the little red triangle is next to line 93.

**Reply by Ssunoo2** [reply, marked as answer]
I believe that perhaps the issue_id was not found because there was not actually any data downloaded. The pathing was correct and I found that using "run all" would result in no data downloaded. I resolved this issue by running each chunk individually, which successfully downloaded the data and the above error did not reappear.
